Modules/socketmodule.h

```diff
@@ -101,39 +101,39 @@ typedef int socklen_t;
 #include <sys/kern_control.h>
 #endif
 
-#ifdef HAVE_SOCKADDR_ALG
-#include <linux/if_alg.h>
-#ifndef AF_ALG
-#define AF_ALG 38
-#endif
-#ifndef SOL_ALG
-#define SOL_ALG 279
-#endif
-
 #ifdef HAVE_LINUX_VM_SOCKETS_H
 # include <linux/vm_sockets.h>
 #else
 # undef AF_VSOCK
 #endif
 
+#ifdef HAVE_SOCKADDR_ALG
+# include <linux/if_alg.h>
+# ifndef AF_ALG
+#  define AF_ALG 38
+# endif
+# ifndef SOL_ALG
+#  define SOL_ALG 279
+# endif
```

Comment:
These `ifdef`s seem rather silly to me. If `linux/if_alg.h` exists, surely `sys/socket.h` is going to define `AF_ALG` and `SOL_ALG`, right?

Reply:
I don't know enough about this topic to be sure, but from a cursory search it appears to be [a guideline in the Linux docs](https://github.com/torvalds/linux/blob/6f0d349d922ba44e4348a17a78ea51b7135965b1/Documentation/crypto/userspace-if.rst#user-space-api-general-remarks) to include this block, in case of older header files that don't yet define these constants themselves.